File: src/main/java/com/isep/acme/domain/service/ImageService.java
package com.isep.acme.domain.service;

import java.util.ArrayList;
import java.util.List;

import org.springframework.core.io.Resource;
import org.springframework.stereotype.Service;

import com.isep.acme.domain.model.ProdImage;
import com.isep.acme.domain.repository.ImageRepository;
import com.isep.acme.dto.ImageDTO;

import lombok.AllArgsConstructor;

@Service
@AllArgsConstructor
public class ImageService {

     private final FileStorageService service;
     private final ImageRepository repository;
     private String filename;

     public Iterable<ImageDTO> getImageProduct() {
          Iterable<ProdImage> p = repository.findAll();
          List<ImageDTO> iDto = new ArrayList<>();
          for (ProdImage pd : p) {
               iDto.add(pd.toDto());
          }
          return iDto;
     };

     public Resource addImage(Resource image) {          
         return service.loadFileAsResource(filename);
     }

}

File: src/domain/service/imageService.ts
import { ImageDTO } from "../../dto/imageDTO";
import { ProdImage } from "../model/prodImage";
import { ImageRepository } from "../repository/imageRepository";
import { FileStorageService } from "./fileStorageService";

export class ImageService {
  private filename: string;

  constructor(
    filename: string,
    private readonly service: FileStorageService,
    private readonly repository: ImageRepository
  ) {
    this.filename =filename
  }

  async getImageProduct(): Promise<ImageDTO[]> {
    const p: ProdImage[] = await this.repository.find();
    const iDto: ImageDTO[] = [];
    for (const pd of p) {
      iDto.push(pd.toDto());
    }
    return iDto;
  }

  // async addImage(image: Resource): Promise<Resource> {
  // return await this.service.loadFileAsResource(this.filename);
  // }
}
